### PR TITLE
Adding index for best practices

### PR DESCRIPTION
**Pull request by garrettjonesgoogle**

### CODE REVIEW

File: library-best-practices/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Java Library Best Practices
+---------------------------
+
+The Java Library Best Practices are a collection of practices that we believe
```
Comment (elharo) [inline library-best-practices/README.md]: "The Java Library Best Practices" --> These

File: library-best-practices/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+will minimize problems for consumers of interconnected Java libraries.  These
+practices come from decades of aggregated experience in maintaining open source
+Java libraries and are informed by many hard-earned lessons from mistakes that
```
Comment (elharo) [inline library-best-practices/README.md]: hard-learned

File: library-best-practices/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+have been made. We have found that following these best practices will result in
```
Comment (elharo) [inline library-best-practices/README.md]: best practices --> rules

File: library-best-practices/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+higher quality Java libraries with fewer diamond dependency conflicts and other
```
Comment (elharo) [inline library-best-practices/README.md]: delete diamond

File: library-best-practices/README.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+kinds of problems. The list is open-ended, so new ones may be added from time to
+time.
+
+- [JLBP-1](JLBP-1.md): Minimize dependencies
+- [JLBP-2](JLBP-2.md): Minimize API surface
+- [JLBP-3](JLBP-3.md): Use Semantic versioning
+- [JLBP-4](JLBP-4.md): Avoid dependencies on unstable libraries and features
+- [JLBP-5](JLBP-5.md): Avoid dependencies that overlap classes with other
```
Comment (elharo) [inline library-best-practices/README.md]: classes or packages?

Comment (garrettjonesgoogle) [reply library-best-practices/README.md]: "overlapping classes" is the way we describe the problem in JLBP-5 and JBLP-6; it's when multiple packages have the same fully-qualified class names.

Comment (elharo) [reply library-best-practices/README.md]: so it's not split packages, then? In that case, we probably do need a new JLBP to "not split packages" and we may want to reread 1-6 in this light. It's particularly important in Java 9 and later where split packages are now banned.

**Comment by garrettjonesgoogle** [comment]
@elharo feedback addressed, PTAL

File: library-best-practices/README.md (hunk 5 above)
Comment (garrettjonesgoogle) [reply library-best-practices/README.md]: Ok makes sense, I'll queue it up.

**Pull request merged**